### PR TITLE
Addressing review comments by Jim Schaad

### PR DESCRIPTION
**Pull request by hannestschofenig**
https://www.ietf.org/mail-archive/web/ace/current/msg02798.html

### CODE REVIEW

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 1]
```diff
@@ -237,22 +233,14 @@
...
-	  the IANA "CBOR Web Token Claims" registry <xref target="IANA.CWT.Claims"/>.
-	</t>
+	The <spanx style="verb">cnf</spanx> claim value may contain one or multiple proof-of-
+    possession keys. Hence, more than one <spanx style="verb">jwk</spanx>, <spanx style="verb">jwe</spanx>, and/or <spanx style="verb">jku</spanx> (JWK
+    Set URL) confirmation values  may be present in a CWT.  A profile may
+ impose
+    further restrictions on the number of PoP keys that are present and
+    also whether all or a only a subset of them needs to be utilized by in
+    a given application domain.
+</t>
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: This change introduces unnecessary syntactic and semantic ambiguity.  It also breaks the semantic equivalence with RFC 7800.  Please remove this change.

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 2]
```diff
@@ -513,7 +501,7 @@
 	This attack can, for example, be avoided when a signed nonce or challenge is used
 	since the recipient can use a distinct nonce or challenge for each interaction.
 	Replay can also be avoided if a sub-key is derived from a shared secret
-	that is specific to the instance of the proof-of-possession demonstration.
+	that is specific to the instance of the proof-of-possession  demonstration.
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: Please remove the extra space probably inadvertently introduced by this change.

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 3]
```diff
@@ -961,9 +949,7 @@
 
     <section title='Acknowledgements' anchor='Acknowledgements' numbered="no">
       <t>
-	Thanks to the following people for their reviews of the specification:
-	Michael Richardson and
-	Jim Schaad.
+	We would like to thank Michael Richardson and Jim Schaad for their reviews of the specification.
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: Please revert this change.  The list of reviewers will grow and it's easier to maintain the list when there's one person per line.

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 4]
```diff
@@ -162,18 +158,22 @@
...
+		Party that creates the CWT and binds the claims about the subject to the 
+		proof-of-possession key.
 	  </t>
 
 	  <t hangText="Presenter">
 	    <vspace/>
 	    Party that proves possession of a private key (for asymmetric key cryptography)
-	    or secret key (for symmetric key cryptography) to a recipient.
+	    or secret key (for symmetric key cryptography) to a recipient.<vspace/>
+		In context of OAuth this party is also called the OAuth Client.
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: Please delete the <vspace/> as it will cause an odd line break between sentences.  Put a comma after OAuth.

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 5]
```diff
@@ -162,18 +158,22 @@
...
 	  </t>
 
 	  <t hangText="Recipient">
 	    <vspace/>
 	    Party that receives the CWT containing the proof-of-possession key information from the presenter.
+		<vspace/>
+		In context of OAuth this party is also called the OAuth Resource Server.
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: Please delete the <vspace/> as it will cause an odd line break between sentences.  Put a comma after OAuth.

File: draft-ietf-ace-cwt-proof-of-possession.xml [hunk 6]
```diff
@@ -207,17 +207,13 @@
...
-	  Other members of the <spanx style="verb">cnf</spanx> map may be defined because
-	  a proof-of-possession key may not be the only means of confirming
-	  the authenticity of the token.
-	  This is analogous to the SAML 2.0 <xref target="OASIS.saml-core-2.0-os"/>
-	  SubjectConfirmation element in which a number of different
-	  subject confirmation methods can be included
-	  (including proof-of-possession key information).
-	</t>
+	  <t>
+  The <spanx style="verb">cnf</spanx> claim in the CWT is used to carry confirmation methods, some of
```
Comment (selfissued) [inline draft-ietf-ace-cwt-proof-of-possession.xml]: This is a run-on sentence.  Either make it two sentences or change the comma to a semicolon.

**Review by selfissued — CHANGES_REQUESTED** [review]
The proposed change to allow multiple PoP keys in the "cnf" element introduces unnecessary syntactic and semantic ambiguity. It also breaks the semantic equivalence with RFC 7800. Please remove this change.

File: draft-ietf-ace-cwt-proof-of-possession.xml (hunk 1 above)
Comment (erdtman) [reply draft-ietf-ace-cwt-proof-of-possession.xml]: +1
If this is really needed it would be good to first have consensus by the WG. If there are no strong use cases I would prefer to keep this limitation since it makes implementing so much easier.

**Comment by erdtman** [comment]
Apart form the number of keys in cnf it looks good to me.

**Comment by hannestschofenig** [comment]
I made a number of changes based on your feedback. I think this should be ready.

**Review by selfissued — APPROVED** [review]
Thanks for the updates, Hannes!
